concordium-consensus/src/Concordium/GlobalState/Persistent/TreeState.hs

```diff
@@ -390,8 +398,11 @@ loadSkovPersistentData rp _treeStateDirectory pbsc = do
     readBlobBSFromHandle (bscBlobStore . PBS.pbscBlobStore $ pbsc) (sbState lfStoredBlock)
   case elfBlob :: Either IOError BS.ByteString of
     Left _ -> logExceptionAndThrowTS $
-      DatabaseInvariantViolation "Last finalized block cannot be read. Blockstate database recovery required."
+        DatabaseInvariantViolation "Last finalized block cannot be read. Blockstate database recovery required."
```

Comment:
This check  does not seem apt anymore.

The only time we'd see this now, it appears to me, is if the state for the genesis block was not written properly. So we could have a more precise error message. Recovery in this case is simple (and cost-free).

Reply:
The intention was not to specifically handle the genesis block state corruption but to indicate that the recovery attempt failed. I cleaned that up now.

The genesis block state recovery will be explicitly handled under the recovery from global state initialisation failure because this is the exact occasion when it can happen.